internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -187,9 +187,34 @@ func (k *KafkaManager) setClusterStatusCapacityMetrics() error {
 		}
 	}
 
+	// enterprise clusters max and available is handled separately
+	enterpriseClustersAvailableAndMaxCapacityCounts := k.calculateAvailableAndMadCapacityForEnterpriseClusters(usedStreamingUnitsCountByRegion)
+	for _, availableAndMaxCapacityCounts := range enterpriseClustersAvailableAndMaxCapacityCounts {
+		k.setClusterStatusCapacityAvailableMetric(availableAndMaxCapacityCounts)
+		k.setClusterStatusCapacityMaxMetric(availableAndMaxCapacityCounts)
+	}
```

Comment:
Is this correct? Are we interested only in the values of the latest element of the array?

Reply:
hi Massimo. can you clarify what you mean?

Reply:
I believe the issue is that you're overwriting the values in every iteration of the loop. Only the last iteration's values are exported.

Reply:
nvm. on second look the code is correct: the `setClusterStatusCapacityMetric` functions take an object which has the region, cloud provider, cluster_id etc.
No values will be overwritten.

Reply:
You call `setClusterStatusCapacityAvailableMetric` for every element in the array.
Unless I'm missing something I'd say you are overwriting the values on each iteration

Reply:
I don't think I am overwriting anything. the loop iterates over each enterprise cluster and sets metrics with the values of that enterprise cluster

Reply:
If those values are 'added' or 'updated' based on some value inside the `availableAndMaxCapacityCounts`, then I believe the method names should be changed to something like `addClusterStatusCapacityAvailableMetric` and `addClusterStatusCapacityMaxMetric`.

Calling a `set` method multiple times makes me think you are updating the same thing multiple times (i.e., overwriting)

Reply:
I get it now. renamed the functions

Reply:
I think there is still some confusion @pawelpaszki @ziccardi 

the original name was `setClusterStatusCapacityAvailableMetric`. The new name is `sendClusterStatusCapacityAvailableMetric`.

What the method does is that it SETS the corresponding cluster status capacity available prometheus metric. If you dig into the code you will see that it is a Prometheus Gauge.
What it is being done is setting the gauge, not sending anything nor adding anything. Prometheus model is that Prometheus scrapes the endpoint. In this case the endpoint is KFM. In KFM we are setting the gauge but not sending it anywhere.
Therefore I think this should still be named setXXXX.

Aside from that, in each iteration we are setting the gauge. This is, overwriting the previous value. This might be intentional or not, depending on what is the meaning of that gauge and what we want to do.

This should be clarified.

Reply:
I reviewed the code and the methods set the same metric name but with different labels so it would be correct

Reply:
I wonder if maybe `update` metrics would be a better prefix than `set`  to make it clear that each call doesn't overwrite the previous one (not always).

Reply:
@miguelsorianod - WDYT about Massimo's suggestion?

Reply:
I am fine with both approaches

Reply:
changed to `update`